Cookbook/Technical-Documents/PlatformTools.md

```diff
@@ -0,0 +1,186 @@
+iOS Platform Tools
+==================
+
+## Overview
+
+This article covers some existing useful dev-tools specifically for iOS Platform Team to improve dependency analysis, build-time measurement, and so on.
+
+See Also:
+- [\[IOSP\-65\] Create the dependency graph](https://babylonpartners.atlassian.net/browse/IOSP-65)
+- [\[IOSP\-65\] Dependency Graph \- iOS 🚀 \- Google Drive](https://drive.google.com/drive/u/1/folders/1XD3emGwNdtbIG60MduGOH9GlepEEsG0w)
+
+---
+
+## Table of Contents
+
+### Dependency Graph Visualiser
+
+- [PaulTaykalo/objc-dependency-visualizer](https://github.com/PaulTaykalo/objc-dependency-visualizer)
+- [matsoftware/swift-code-metrics](https://github.com/matsoftware/swift-code-metrics)
+
+### Build Time Measurement
+
+- [PaulTaykalo/xcode-build-times-rendering](https://github.com/PaulTaykalo/xcode-build-times-rendering)
+- [spotify/XCLogParser](https://github.com/spotify/XCLogParser)
+
+### Others
+
+- [uber/uber-poet](https://github.com/uber/uber-poet)
+
+---
+
+## Dependency Graph Visualiser
+
+### 🛠 [PaulTaykalo/objc-dependency-visualizer](https://github.com/PaulTaykalo/objc-dependency-visualizer)
+
+Guide: [Home · PaulTaykalo/objc\-dependency\-visualizer Wiki](https://github.com/PaulTaykalo/objc-dependency-visualizer/wiki)
+
+`objc-dependency-visualizer` is written in Ruby which searches for Xcode-generated `.swiftdeps` files in `DerivedData` to collect all dependencies to create a beautiful graph visualisation using `d3.js` .
+
+Each node corresponds to either type or global variable / function.
+
+DOT format with graphviz rendering is also supported.
+
+#### Output
+
+<img src="Assets/platform-tooling/visualizer-Octopus-ExchangeableTabBar.png">
+<img src="Assets/platform-tooling/visualizer-BabylonDependencies.png">
+<img src="Assets/platform-tooling/visualizer-BabylonCore-graphviz.png">
+
+#### Script
+
+```sh
```

Comment:
Should we have the scripts as files ready to run instead of putting the code in here?

Reply:
It would probably be nice to have CI reporting periodically with scripts and tools managed in some repo,
but at the moment, this is only for the research, so it might be too early to prepare such environment (we want to make sure it will be well-maintained).

But some of the tools can also report in JSON / DOT format that can detect the (unexpected) dependency-graph changes, so such notifier might be worth implementing in future.